File: plugins/dingxiang/dingxiang.py
import logging
from handle_excel import ExcelHandle
import time
import traceback


class CustomExcelHandle(ExcelHandle):

    def __init__(self, name, upload_file):
        super().__init__(name, upload_file)
        self.custom_name = "丁香（无痕）"

    def start_handle_excel(self):
        logging.info("{} start handle excel".format(self.name))
        wb, ws, items = self.init_read_excel()
        try:
            for index, row in enumerate(items):
                sn = row[2].value
                shop_sn = row[2].value
                receive_people = row[19].value
                phone = row[20].value
                province = row[21].value
                city = row[22].value
                county = row[23].value
                address = row[24].value
                sales_channel_name = self.custom_name
                product_name = row[10].value
                barcode = row[13].value
                numbers = row[15].value
                unit_price = row[18].value
                ws.append([sn, shop_sn, "", "", "", "", self.custom_name, "", "", receive_people, phone, "", "",
                           province, city, county, address, "", "", "", "", "", "", "", "", sales_channel_name, "",
                           product_name, barcode, "", "", "", numbers, unit_price, "", "", "", "", "", "", "", "", "",
                           "", "", "", "", "", "", "", ""])
            result_file_path = "./result/dingxiang/丁香{0}.xlsx".format("".join(str(time.time()).split(".")))
            wb.save(result_file_path)
            logging.info("{0} handle excel file {1} success".format(self.name, self.upload_file))

        except Exception as e:
            exc = traceback.format_exc()
            logging.error("error %s" % exc)
        wb.close()
        self.delete_success_file()

File: plugin/dingxiang/dingxiang.go
package dingxiang

import (
	"fmt"
	"github.com/peanut-cc/peanut_case/plugin"
	"github.com/peanut-cc/peanut_case/utils/uuid"
	"github.com/xuri/excelize/v2"
	"log"
	"os"
)

func init() {
	dingxiang := &DingXiang{Name: plugin.DingXiang, CustomName: "无痕-杭州丁香健康管理有限公司（丁香）"}
	plugin.PluginMap[plugin.DingXiang] = dingxiang
}

type DingXiang struct {
	Name       string
	CustomName string
}

func (p *DingXiang) GetPluginName() string {
	return p.Name
}

func (p *DingXiang) HandleUploadFile(fileName string) error {
	rows, err := plugin.ReadExcel(fileName)
	if err != nil {
		fmt.Printf("客户{%v} 打开excel 报错{%v}", p.GetPluginName(), err)
		return err
	}
	f := excelize.NewFile()
	sheet := f.NewSheet("Sheet1")
	sheetName := f.GetSheetName(sheet)
	err = f.SetSheetRow(sheetName, "A1", &plugin.RowHeader)
	if err != nil {
		log.Printf("客户 {%v} 创建excel失败\n", p.Name)
		return err
	}

	for index, row := range rows {
		// 第一行的标题不做处理
		if index == 0 {
			continue
		}
		sn := row[2]
		shopSn := row[2]
		receivePeople := row[19]
		phone := row[20]
		province := row[21]
		city := row[22]
		county := row[23]
		address := row[24]
		salesChannelName := p.CustomName
		productName := row[10]
		barcode := row[13]
		numbers := row[15]
		unitPrice := row[18]
		axis := fmt.Sprintf("A%d", index+1)
		err = f.SetSheetRow(sheetName, axis, &[]string{sn, shopSn, "", "", "", "", p.CustomName, "", "", receivePeople, phone, "", "",
			province, city, county, address, "", "", "", "", "", "", "", "", salesChannelName, "",
			productName, barcode, "", "", "", numbers, unitPrice, "", "", "", "", "", "", "", "", "",
			"", "", "", "", "", "", "", ""})
		if err != nil {
			log.Printf("客户 {%v} 写excel 第 {%v} 错误\n", p.Name, index)
			return err
		}
	}
	filename := fmt.Sprintf("./result/丁香/丁香%v.xlsx", uuid.MustString())
	err = f.SaveAs(filename)
	if err != nil {
		log.Printf("保存{%v} 失败\n", filename)
		return err
	}
	if err := p.DeleteUploadFile(fileName); err != nil {
		log.Printf("删除{%v} 失败\n", fileName)
		return err
	}
	log.Printf("客户{%v} 订单{%v}处理完毕\n", p.Name, fileName)
	return nil
}

func (p *DingXiang) DeleteUploadFile(fileName string) error {
	return os.Remove(fileName)
}
